jdaviz/app.py

```diff
@@ -109,72 +109,76 @@ def to_unit(self, data, cid, values, original_units, target_units):
                 spec = data.get_object(cls=Spectrum1D)
 
             except RuntimeError:
-                eqv = []
+                data = data.get_object(cls=NDDataArray)
+                spec = Spectrum1D(flux=data.data * u.Unit(original_units))
```

Comment:
Should we pass along the attributes for spectral axis, meta, or WCS, too? There are places within the methods below that call Spectrum1D's `spectral_axis` and `meta`, so we might need to make sure those get passed in here too.

Reply:
What value should I pass for the `spectral_axis` parameter? Otherwise yes that makes sense.

Reply:
Good question. We could pull it out of the `wcs` by getting out the spectral component of the coordinates, and evaluate it on the integer pixel grid.

Reply:
Sure thing 💯